src/components/Nav.jsx

```diff
@@ -33,14 +36,14 @@ class Nav extends Component {
         this.state.settings.busy ||
         this.state.settings.opened
         ) {
-        // if it's not a cozy-bar nav popup, close the opened popup
-        if (!this.rootRef.contains(event.target)) {
-          this.setState({ // reset all
-            apps: {busy: false, opened: false},
-            settings: {busy: false, opened: false}
-          })
-        }
-        event.stopPropagation()
+      // if it's not a cozy-bar nav popup, close the opened popup
```

Comment:
Cannot this part be in the Modal component ? Like "any modal close any other opened modal" ?

Reply:
I think not because the common state is handled in this component. If I put in the modal component, it won't know that another modal is opened, I think.

Reply:
Maybe with a `static` property ? What thinks @goldoraf about it ?

Reply:
Well, I'm not a big fan of `static` properties, even if I used then first in the Notifier. I recently tried another approach related to modal management, using redux of course: https://github.com/cozy/cozy-drive/blob/master/src/lib/react-cozy-helpers/ModalManager.jsx

The idea is to attach the ModalManager in your layout (https://github.com/cozy/cozy-drive/blob/master/src/drive/components/Layout.jsx#L22) and to dispatch a `showModal` action like this: https://github.com/cozy/cozy-drive/blob/master/src/drive/ducks/files/Container.jsx#L89

You could easily enhance the ModalManager so that it could handle multiple modals ;)

Reply:
That seems a good idea indeed, but for an application only. Since the bar don't and won't use redux I would prefer to keep the current way which seems to be better and simpler.